Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocController.swift

```diff
@@ -0,0 +1,470 @@
+
+import Foundation
+import UIKit
+import CocoaLumberjackSwift
+
+protocol ArticleAsLivingDocControllerDelegate: class {
+    var articleURL: URL { get }
+    var article: WMFArticle { get }
+    var messagingController: ArticleWebMessagingController { get }
+    var theme: Theme { get }
+    var webView: WKWebView { get }
+    var leadImageContainerView: UIView { get }
+    func updateArticleMargins()
+}
+
+@available(iOS 13.0, *)
+class ArticleAsLivingDocController: NSObject {
+    
+    enum Errors: Error {
+        case viewModelInstantiationFailure
+    }
+    
+    private weak var delegate: (ArticleAsLivingDocControllerDelegate & UIViewController & HintPresenting)? //will need ArticleAsLivingDocViewControllerDelegate
+    
+    var _articleAsLivingDocViewModel: ArticleAsLivingDocViewModel?
```

Comment:
Yeah it is a bit messy here, agreed. I'm going to leave it as-is for now since it feels a little more impactful, but I'll make a note to revisit with the modal code in place (so it's easier to test against).